execution_engine/src/system/auction.rs

```diff
@@ -528,93 +528,93 @@ pub trait Auction:
 
     /// Mint and distribute seigniorage rewards to validators and their delegators,
     /// according to `reward_factors` returned by the consensus component.
-    fn distribute(&mut self, reward_factors: BTreeMap<PublicKey, u64>) -> Result<(), Error> {
+    // TODO: rework EraInfo and other related structs, methods, etc. to report correct era-end
+    // totals of per-block rewards
+    fn distribute(&mut self, proposer: PublicKey) -> Result<(), Error> {
         if self.get_caller() != PublicKey::System.to_account_hash() {
             return Err(Error::InvalidCaller);
         }
 
+        // Workaround for the nominal proposer of the genesis block being the system account
+        if proposer == PublicKey::System {
+            return Ok(());
+        }
+
         let seigniorage_recipients = self.read_seigniorage_recipients()?;
         let base_round_reward = self.read_base_round_reward()?;
         let era_id = detail::get_era_id(self)?;
 
         let mut era_info = EraInfo::new();
         let seigniorage_allocations = era_info.seigniorage_allocations_mut();
 
-        for (public_key, reward_factor) in reward_factors {
-            let recipient = seigniorage_recipients
-                .get(&public_key)
-                .ok_or(Error::ValidatorNotFound)?;
+        let recipient = seigniorage_recipients
+            .get(&proposer)
+            .ok_or(Error::ValidatorNotFound)?;
 
-            let total_stake = recipient.total_stake().ok_or(Error::ArithmeticOverflow)?;
-            if total_stake.is_zero() {
-                // TODO: error?
-                continue;
-            }
+        let total_stake = recipient.total_stake().ok_or(Error::ArithmeticOverflow)?;
+        /*if total_stake.is_zero() {
+            // TODO: error?
+            continue;
+        }*/
```

Comment:
I guess this should be unreachable anyway, and even if, it would do the right thing (compute 0 rewards) anyway?
I'd prefer to either keep or remove the code, rather than commenting it out.